core/controllers/editor.py

```diff
@@ -691,9 +714,22 @@ def put(self, exploration_id):
                 exp_domain.ExplorationChange(change)
                 for change in change_list_dict]
             version = self.payload.get('version')
-            exp_services.create_or_update_draft(
-                exploration_id, self.user_id, change_list, version,
-                datetime.datetime.utcnow())
+
+            exploration_rights = rights_manager.get_exploration_rights(
```

Comment:
Line 718 to 732 looks duplicate of 147-159, if so can you please create a function which can handle this and remove duplicate code? (You can do it in a separate PR  (create an issue and assign it to yourself), as this PR will create merge conflicts in your other PRs.)

Reply:
I think you can do it in the next commit.

Reply:
@DubeySandeep, the duplicate code here is from line `718 to 722` and line `147 to 152`. The other lines are specific to the handler